web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -155,12 +155,41 @@ export class LayerDialogComponent implements OnDestroy {
     this.projectId = project.id;
   }
 
-  getForm(formId: string, fields: Map<string, Field>) {
-    const form = {
-      id: formId,
-      fields,
-    };
-    return form;
+  convertQuersionToField(fieldId: string, question: Question): Field {
+    if (question.fieldType.type === 'text') {
+      return new Field(
+        fieldId,
+        FieldType['TEXT'],
+        StringMap({
+          en: question.label || '',
+        }),
+        /*required=*/ false,
+        /*multipleChoice=*/ undefined
+      );
+    }
+    if (question.fieldType.type === 'multipleChoice') {
+      let options = Map<string, Option>();
+      question.options.forEach((option: OptionModel) => {
+        const optionId = this.dataStoreService.generateId();
+        options = options.set(optionId, {
+          id: optionId,
+          code: option.code || '',
+          label: StringMap({
+            en: option.label || '',
+          }),
+        });
+      });
+      return new Field(
```

Comment:
I agree to your point, I will update the data store function to iterate fields and convert mapped object to json. I just mentioned this because I was not able to save form when I ran your branch

Reply:
haha, well, i broke it my branch so i should fix it, no?;) 

It would be nice though if there was an actual unit test that is testing this behaviour so lazy people like me to forgot to change it by hand do not mess it up. 

Do you think that you could add a test?

Reply:
hahah, yes you can fix it that would be awesome. I would add a unit test wdyt?

Reply:
ok! I made it convert back and was able to successfully create new layer